FamilySearchIosSampleApp/LoginVC.swift

```diff
@@ -0,0 +1,244 @@
+    @IBOutlet weak var dataUsageLabel: UILabel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        activityIndicator.stopAnimating()
+        
+        usernameTextFiew.placeholder = NSLocalizedString("usernamePlaceholderText", comment: "username, in email form")
+        passwordTextField.placeholder = NSLocalizedString("passwordPlaceholderText", comment: "password")
+        loginButtonOutlet.setTitle(NSLocalizedString("loginText", comment: "text for login button"), forState: UIControlState.Normal)
+        dataUsageLabel.text = NSLocalizedString("loginDataUsage", comment: "description of data usage")
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    @IBAction func loginAction(sender: AnyObject)
+    {
+        activityIndicator.startAnimating()
+        
+        let username = usernameTextFiew.text!
+        let password = passwordTextField.text!
+
+        if ( !(username.isEmpty) && !(password.isEmpty))
+        {
+            // get initial GET call to collections
+            Utilities.getUrlsFromCollections({ (collectionsResponse, error) -> Void in
+                if (error == nil)
+                {
+                    // get the login token
+                    self.getToken(collectionsResponse.tokenUrlString!,
+                        username: username,
+                        password: password,
+                        client_id: AppKeys.API_KEY,
+                        completionToken: {(responseToken, errorToken) -> Void in
+                            if (errorToken != nil)
+                            {
+                                // TODO: handle case when somehow the token is nil
+                            }
+                            else
+                            {
+                                // get user data, with the newly acquired token
+                                self.getCurrentUserData(collectionsResponse.currentUserString!,
+                                    accessToken: responseToken!,
+                                    completionCurrentUser:{(responseUser, errorUser) -> Void in
+                                        if (errorUser != nil)
+                                        {
+                                            // TODO: handle case when somehow the user data is nil
+                                        }
+                                        else
+                                        {
+                                            // all login data needed has been downloaded
+                                            // push to the next view controller, in the main thread
+                                            dispatch_async(dispatch_get_main_queue(),{
+                                                self.performSegueWithIdentifier("segueToTabBar", sender: responseUser)
```

Comment:
Are you suggesting I change this:

`dispatch_async(dispatch_get_main_queue(),{
    self.performSegueWithIdentifier("segueToTabBar", sender: responseUser)
})`

with this:
`dispatch_async(dispatch_get_main_queue(),{
    [unowned self] in
    self.performSegueWithIdentifier("segueToTabBar", sender: responseUser)
})`